ios/RNSScreenStack.mm

```diff
@@ -735,43 +737,8 @@ - (void)cancelTouchesInParent
   // item is close to an edge and we start pulling from edge we want the Touchable to be cancelled.
   // Without the below code the Touchable will remain active (highlighted) for the duration of back
   // gesture and onPress may fire when we release the finger.
-#ifdef RCT_NEW_ARCH_ENABLED
-  // On Fabric there is no view that exposes touchHandler above us in the view hierarchy, however it is still
-  // utilised. `RCTSurfaceView` should be present above us, which hosts `RCTFabricSurface` instance, which in turn
-  // hosts `RCTSurfaceTouchHandler` as a private field. When initialised, `RCTSurfaceTouchHandler` is attached to the
-  // surface view as a gestureRecognizer <- and this is where we can lay our hands on it.
-  UIView *parent = _controller.view;
-  while (parent != nil && ![parent isKindOfClass:RCTSurfaceView.class]) {
-    parent = parent.superview;
-  }
 
-  // This could be possible in modal context
-  if (parent == nil) {
-    return;
-  }
-
-  RCTSurfaceTouchHandler *touchHandler = nil;
-  // Experimentation shows that RCTSurfaceTouchHandler is the only gestureRecognizer registered here,
-  // so we should not be afraid of any performance hit here.
-  for (UIGestureRecognizer *recognizer in parent.gestureRecognizers) {
-    if ([recognizer isKindOfClass:RCTSurfaceTouchHandler.class]) {
-      touchHandler = static_cast<RCTSurfaceTouchHandler *>(recognizer);
-    }
-  }
-
-  [touchHandler rnscreens_cancelTouches];
-#else
-  // On Paper we can access touchHandler hosted by `RCTRootContentView` which should be above ScreenStack
-  // in view hierarchy.
-  UIView *parent = _controller.view;
-  while (parent != nil && ![parent respondsToSelector:@selector(touchHandler)]) {
-    parent = parent.superview;
-  }
-  if (parent != nil) {
-    RCTTouchHandler *touchHandler = [parent performSelector:@selector(touchHandler)];
-    [touchHandler rnscreens_cancelTouches];
-  }
-#endif // RCT_NEW_ARCH_ENABLED
+  [[self rnscreens_findTouchHandlerInAncestorChain] rnscreens_cancelTouches];
```

Comment:
Can we do something like:

```obj-c
RNS_TOUCH_HANDLER_ARCH_TYPE *touchHandler = [self rnscreens_findTouchHandlerInAncestorChain];
[touchHandler rnscreens_cancelTouches];
```

to improve readability?

Reply:
I find it quite opposite 😄 I'm mean the redundant variable strikes me => I'll stay by current version as this is taste related, but thanks.